cmd/dockerd/daemon_linux.go

```diff
@@ -35,6 +35,37 @@ func notifyReady() {
 	go systemdDaemon.SdNotify(false, systemdDaemon.SdNotifyReady)
 }
 
+// notifyReloading sends a message to the host when the server got signaled to
+// reloading its configuration, see [sd_notify(3)]. [notifyReady] must be called
+// after reloading finishes. The server should be running as a systemd unit with
+// "Type=notify" (see [systemd.service(5)]).
+//
+// Note: we currently use the pre-systemd 253 implementation, which uses "Type=notify",
+// combined with [ExecReload]. The [ExecReload] option is designed to be synchronous,
+// which complicates its use when signals (SIGHUP) is used to reload:
+//
+// > Note however that reloading a daemon by enqueuing a signal (...) is usually
+// > not a good choice, because this is an asynchronous operation and hence not
+// > suitable when ordering reloads of multiple services against each other.
+//
+// Systemd 253 introduced a new Type (Type=notify-reload, see [systemd#25916]),
+// which allows setting a "ReloadSignal" instead, in addition to sending a
+// [MONOTONIC_USEC].
+//
+// We currently still support distros that do not provide systemd 253, so cannot
+// use this new feature, but sending "RELOADING=1 / "READY=1" should at least
+// provide more information to systemd for the time being.
+//
+// [sd_notify(3)]: https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html#RELOADING=1
+// [systemd.service(5)]: https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#Type=
+// [ExecReload]: https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#ExecReload=
+// [MONOTONIC_USEC]: https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html#MONOTONIC_USEC=…
+// [systemd#25916]: https://github.com/systemd/systemd/pull/25916
+func notifyReloading() {
+	// TODO(thaJeztah): use "Type=notify-reload", "ReloadSignal", and "MONOTONIC_USEC" once we drop support for systemd < 253, and supported by github.com/coreos/go-systemd.
+	go systemdDaemon.SdNotify(false, systemdDaemon.SdNotifyReloading)
```

Comment:
```suggestion
	systemdDaemon.SdNotify(false, systemdDaemon.SdNotifyReloading)
```

Ditto for `notifyReady()`. Notifying asynchronously makes it possible for the two notifications to race, and possibly get delivered to systemd out of order.

Reply:
> Notifying asynchronously makes it possible for the two notifications to race, and possibly get delivered to systemd out of order.

I was considering that as well yeah. Do you think we should call them synchronously?